File: frontend/src/app/models/TermsAndCondition.ts
import { BaseModel } from './BaseModel';

export interface TermsAndConditionDto extends BaseModel {
  title: string;
  description: string;
  selected?: boolean;
}

export interface CreateTermsAndConditionRequest {
  title: string;
  description: string;
}

export interface UpdateTermsAndConditionRequest extends CreateTermsAndConditionRequest {
  id: string;
}

export interface TermItem {
  title: string;
  description?: string;
}

export interface CreateTermsAndConditionBulkRequest {
  items: TermItem[];
}

File: backend/Model/TermsAndCondition.cs
using System.ComponentModel.DataAnnotations;

namespace YLWorks.Model
{
    public class TermsAndCondition : BaseEntity
    { 
        public Guid Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
    }

    public class CreateTermsAndConditionRequest
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
    }

    public class UpdateTermsAndConditionRequest
    {
        public Guid Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
    }

    public class TermItem
    {
        public string Title { get; set; }
        public string? Description { get; set; }
    }

    public class CreateTermsBulkRequest
    {
        public List<TermItem> Items { get; set; } = new();
    }
}